low-level/framework/utils/salt_util.py

```diff
@@ -13,88 +13,114 @@
 # about this software or licensing, please email opensource@seagate.com or
 # cortx-questions@seagate.com.
 
-import subprocess
+
+'''Module which provides common config information using Provisioner API'''
+
 
 try:
-    from utility import Utility
     from service_logging import logger
-except Exception as e:
-    from framework.utils.utility import Utility
+    from utility import Utility
+except Exception as err:
     from framework.utils.service_logging import logger
+    from framework.utils.utility import Utility
 
-class SaltInterface:
+
+class CommonConfig:
     """
-    Interface class to run salt commands and get values from cluster.sls
+    Interface class to get common config from common pillar using
+    Provisioner API
     """
 
+    __instance = None
+    CLUSTER_KEY = 'cluster'
+    CLUSTER_TYPE_KEY = 'type'
+    SSPL_KEY = 'sspl'
+    DATASTORE_KEY = 'DATASTORE'
+
     def __init__(self):
-        self.GRAINS_GET_NODE_CMD = "sudo salt-call grains.get id --output=newline_values_only"
-        self.MINION_GET_NODE_CMD = "cat /etc/salt/minion_id"
-        self.utility = Utility()
-
-    def get_node_id(self):
-        # TODO : need to fetch node_key using salt python API.
-        # Facing issue of service is going in loop till it eat's all the memory
-        node_key = 'srvnode-1'
+        '''init method'''
+        if CommonConfig.__instance is None:
+            self.utility = Utility()
+            self.pillar_info = None
+            _result, _err, _ret_code = \
+                    self.utility.execute_cmd(['sudo', 'provisioner', 'pillar_get'])
+            _err = _err.decode("utf-8").rstrip()
```

Comment:
Is _err a local variable ? I see it is referenced outside of this function as well.

Reply:
Fixed this problem now